### ISSUE TITLE
Why there is no nodeSelector in spec level or in Job in CRD?

### ISSUE THREAD

**Issue by ChethanUK** [GoogleCloudPlatform/flink-on-k8s-operator#380]
Why there is no nodeSelector in spec level or in Job in latest [crd](https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/docs/crd.md)?

Having node selector in Job also will be helpful or at `spec` level will be much better

Also, there is no option to set [affinity and anti affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity)

**Comment by yanghui16355** [comment]
+1 for including affinity and anti affinity in CRD

**Comment by functicons** [comment]
Yeah, that's something we missed in the past, it should be added.

**Comment by functicons** [comment]
@ChethanUK Do you want to make a PR for it?

**Comment by the-cybersapien** [comment]
Is this still open? Would like to work on this if it's not taken by someone else yet.

**Comment by ChethanUK** [comment]
@the-cybersapien I got busy with something else, I missed this. You can pick up this.

**Comment by sgraham785** [comment]
created a PR #423 for this, as well as #422 which adds affinity to the jobmanager & taskmanager specs. 
apologies if I stepped on anyone's toes here
